files/setup_freeipa.sh

```diff
@@ -0,0 +1,127 @@
+    ip --family inet address show dev "$1" | \
+        grep inet | \
+        sed "s/^ *//" | \
+        cut --delimiter=' ' --fields=2 | \
+        cut --delimiter='/' --fields=1
+}
+
+# Install FreeIPA as a server or replica
+function setup {
+    case "$1" in
+        master)
+            # Grab our IP address
+            interface=$(get_interface)
+            ip_address=$(get_ip "$interface")
+
+            # Install the master
+            #
+            # realm, domain, and hostname are defined in the FreeIPA
+            # variables file that is sourced toward the top of this
+            # file.  Hence we can ignore the "undefined variable"
+            # warnings from shellcheck.
+            #
+            # shellcheck disable=SC2154
+            ipa-server-install --setup-kra \
+                               --realm="$realm" \
+                               --domain="$domain" \
+                               --hostname="$hostname" \
+                               --ip-address="$ip_address" \
+                               --no-ntp \
+                               --no_hbac_allow \
+                               --mkhomedir
+
+            # Get kerberos credentials and create the dhs_certmapdata
+            # rule
+            kinit admin
+            ipa certmaprule-add dhs_certmapdata \
+                --matchrule '<ISSUER>O=U.S. Government' \
+                --maprule '(ipacertmapdata=X509:<I>{issuer_dn!nss_x500}<S>{subject_dn!nss_x500})'
+            ;;
+        replica)
+            # Install the replica
+            #
+            # For some reason ipa-server-install does not appear to
+            # pass the principal to ipa-client-install, so we first
+            # run ipa-client-install manually.
+            #
+            # hostname is defined in the FreeIPA variables file that
+            # is sourced toward the top of this file.  Hence we can
+            # ignore the "undefined variable" warning from shellcheck.
+            #
+            # shellcheck disable=SC2154
+            ipa-client-install --hostname="$hostname" \
+                               --mkhomedir \
+                               --no-ntp
+            ipa-replica-install --setup-ca \
+                                --setup-kra
+            ;;
+        *)
+            echo "Unknown installation type.  Valid installation types are: master | replica"
+            exit 255
+            ;;
+    esac
+
+    # Add the DHS CA to the pkinit anchors
+    sed -i \
+        "/pkinit_anchors = FILE:\/var\/kerberos\/krb5kdc\/cacert\.pem/a \ \ pkinit_anchors = FILE:/usr/local/share/dhsca_fullpath.pem" \
+        /var/kerberos/krb5kdc/kdc.conf
+    systemctl restart krb5kdc.service
+
+    # Add a principal alias for the instance ID so folks can ssh in
+    # via SSM Session Manager.
+    ipa host-add-principal \
+        "$hostname" \
+        host/"$(curl http://169.254.169.254/latest/meta-data/instance-id)"."$domain"
+}
+
+
+if [ $# -lt 1 ]
```

Comment:
Wouldn't `if [ $# -ne 1 ]` be better here?

Reply:
See commit 372efa8.